blob/drivertest/drivertest.go

```diff
@@ -317,6 +319,96 @@ func testList(t *testing.T, newHarness HarnessMaker) {
 			}
 		})
 	}
+
+	// Verify pagination works when inserting in a retrieved page.
+	t.Run("PaginationConsistencyAfterInsert", func(t *testing.T) {
+		drv, done := init(t)
+		defer done()
+
+		// Fetch a page of 2 results: 0, 1.
+		page, err := drv.ListPaged(ctx, &driver.ListOptions{
+			PageSize: 2,
+			Prefix:   keyPrefix,
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		got := gotIndices(t, page.Objects)
+		want := []int{0, 1}
+		if diff := cmp.Diff(got, want); diff != "" {
+			t.Fatalf("got\n%v\nwant\n%v\ndiff\n%s", got, want, diff)
+		}
+
+		// Insert a key "0a" in the middle of the page we already retrieved.
+		b := blob.NewBucket(drv)
+		key := page.Objects[0].Key + "a"
+		if err := b.WriteAll(ctx, key, content, nil); err != nil {
+			t.Fatal(err)
+		}
+		defer func() {
+			_ = b.Delete(ctx, key)
+		}()
+
+		// Fetch the next page. It should not include 0, 0a, or 1, and it should
+		// include 2.
+		page, err = drv.ListPaged(ctx, &driver.ListOptions{
+			Prefix:    keyPrefix,
+			PageToken: page.NextPageToken,
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		got = gotIndices(t, page.Objects)
+		want = []int{2}
+		if diff := cmp.Diff(got, want); diff != "" {
+			t.Errorf("got\n%v\nwant\n%v\ndiff\n%s", got, want, diff)
+		}
```

Comment:
In this case, do you define the behavior of calling ListPaged again on the newly returned page.NextPageToken? Should it return {0a}?

Reply:
What do you mean by the newly returned NextPageToken?

- Do you mean calling `ListPaged` again with the same arguments as before? If so -- it should returned the same thing, just `2` again.
- Do you mean calling `ListPaged` with the `NextPageToken` returned by the last called to `ListPaged`? If so, it will be empty -- there are no more results.

If the concrete type called `ListPaged` again with an empty `PageToken`, it would start iteration again at the beginning, returning 0, 0a, 1, 2.... (with pagination).